lib/matplotlib/tight_bbox.py

```diff
@@ -23,22 +25,24 @@ def adjust_bbox(fig, bbox_inches, fixed_dpi=None):
 
     fig.set_tight_layout(False)
 
-    asp_list = []
     locator_list = []
     for ax in fig.axes:
         pos = ax.get_position(original=False).frozen()
         locator_list.append(ax.get_axes_locator())
-        asp_list.append(ax.get_aspect())
 
         def _l(a, r, pos=pos):
             return pos
         ax.set_axes_locator(_l)
-        ax.set_aspect("auto")
+        # override the method that enforces the aspect ratio
+        # on the Axes
+        ax.apply_aspect = no_op_apply_aspect
 
     def restore_bbox():
-        for ax, asp, loc in zip(fig.axes, asp_list, locator_list):
-            ax.set_aspect(asp)
+        for ax, loc in zip(fig.axes, locator_list):
             ax.set_axes_locator(loc)
+            # delete our no-op function which un-hides the
+            # original method
+            del ax.apply_aspect
```

Comment:
I'd like to get confirmation that this covers all the cases that #17561 is supposed to now handle, e.g., overridden methods, etc..

Reply:
I think it now handles everything.  If it does not exist, things are super broken so we only have to handle `apply_aspect` being in the instance dict or not.

We don't have anything in the test suite that exercises the "in instance dict" code path in the test suite.